src/java/picard/util/ClippingUtility.java

```diff
@@ -170,12 +170,15 @@ public static AdapterPair adapterTrimIlluminaPairedReads(final SAMRecord read1,
                     getReadBases(read1), adapterPair.get3PrimeAdapterBytes(), minMatchBases, maxErrorRate);
             final int index2 = findIndexOfClipSequence(
                     getReadBases(read2), adapterPair.get5PrimeAdapterBytesInReadOrder(), minMatchBases, maxErrorRate);
+            final int numBasesClipped1 = (NO_MATCH == index1) ? 0 : read1.getReadLength() - index1;
+            final int numBasesClipped2 = (NO_MATCH == index2) ? 0 : read2.getReadLength() - index2;
 
-            if (index1 == index2) {
+            // Be careful to count the number of bases clipped, rather than where in the read, since the read lengths may differ.
+            if (numBasesClipped1 == numBasesClipped2) {
```

Comment:
I think this change is wrong.  The point here is that if you have two reads that are reading towards each other's start point, through the template, that they should both encounter adapter sequence after the same number of bases (i.e. the template or insert length), and so finding the adapter after the same number of bases in both reads is a very strong signal that you've got it right.